--- a/arch/arm/boot/dts/imx23-olinuxino.dts	2016-02-25 21:01:36.000000000 +0100
+++ b/arch/arm/boot/dts/imx23-olinuxino.dts	2016-02-27 15:07:31.357127338 +0100
@@ -66,6 +66,14 @@
 				pinctrl-names = "default";
 				pinctrl-0 = <&spi2_pins_a>;
 				status = "okay";
+
+			    eth1: enc28j60@0 {
+					reg = <0>;
+					compatible = "microchip,enc28j60";
+					spi-max-frequency = <1000000>;
+					interrupt-parent = <&gpio0>;
+					interrupts = <2 0x02>;
+				};
 			};
 		};
 
